Sources/SwiftSyntax/RawSyntax.swift

```diff
@@ -767,18 +790,36 @@ extension RawSyntax {
       descendantCount += node.totalNodes
       arena.addChild(node.arenaReference)
     }
+    var recursiveFlags = RecursiveRawSyntaxFlags()
+    if kind.hasError || layoutBuffer.contains(where: { $0?.hasError ?? false }) {
+      recursiveFlags.insert(.hasError)
+    }
```

Comment:
Please do this in the loop above
```swift
var recursiveFlags = RecursiveRawSyntaxFlags()
for case let node? in layoutBuffer {
  byteLength += node.byteLength
  descendantCount += node.totalNodes
  if node.hasError {
    recursiveFlags.insert(.hasError)
  }
  arena.adChild(node.arenaReference)
}
```

Reply:
Done. Also changed `RawSyntax` to have `recursiveFlags` property instead of `hasError` such that we can just do `recursiveFlags.insert(node.recursiveFlags)`, which will also work for all new recursive flags.